plugins/com.google.cloud.tools.eclipse.appengine.libraries/src/com/google/cloud/tools/eclipse/appengine/libraries/repository/MavenHelper.java

```diff
@@ -0,0 +1,87 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.eclipse.appengine.libraries.repository;
+
+import com.google.cloud.tools.eclipse.appengine.libraries.Messages;
+import com.google.cloud.tools.eclipse.appengine.libraries.model.MavenCoordinates;
+import com.google.cloud.tools.eclipse.util.MavenUtils;
+import com.google.cloud.tools.eclipse.util.status.StatusUtil;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Collections;
+import java.util.List;
+import org.apache.maven.artifact.Artifact;
+import org.apache.maven.artifact.repository.ArtifactRepository;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IPath;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.Path;
+import org.eclipse.osgi.util.NLS;
+
+public class MavenHelper {
+
+  public Artifact resolveArtifact(MavenCoordinates mavenCoordinates,
+                                  IProgressMonitor monitor) throws CoreException {
+    List<ArtifactRepository> repository = getRepository(mavenCoordinates);
+    return MavenUtils.resolveArtifact(monitor,
+                                      mavenCoordinates.getGroupId(),
+                                      mavenCoordinates.getArtifactId(),
+                                      mavenCoordinates.getType(),
+                                      mavenCoordinates.getVersion(),
+                                      mavenCoordinates.getClassifier(),
+                                      repository);
+  }
+
+  public IPath getMavenSourceJarLocation(Artifact artifact,
+                                         IProgressMonitor monitor) {
+    try {
+      MavenCoordinates sourceMavenCoordinates = new MavenCoordinates(artifact);
+      sourceMavenCoordinates.setClassifier("sources");
+      Artifact sourceArtifact = resolveArtifact(sourceMavenCoordinates, monitor);
+      return new Path(sourceArtifact.getFile().getAbsolutePath());
+    } catch (CoreException exception) {
+      // source file failed to download, this is not an error
+      return null;
+    }
+  }
+  
+  private List<ArtifactRepository> getRepository(MavenCoordinates mavenCoordinates)
+                                                                              throws CoreException {
+    if (MavenCoordinates.MAVEN_CENTRAL_REPO.equals(mavenCoordinates.getRepository())) {
+      // M2Eclipse will use the Maven Central repo in case null is used
+      return null;
+    } else {
+      return Collections.singletonList(getCustomRepository(mavenCoordinates.getRepository()));
+    }
+  }
+
+  private ArtifactRepository getCustomRepository(String repository) throws CoreException {
+    try {
+      URI repoUri = new URI(repository);
+      if (!repoUri.isAbsolute()) {
+        throw new CoreException(StatusUtil.error(this, NLS.bind(Messages.RepositoryUriNotAbsolute,
```

Comment:
the `this` could be replaced with the class, but this is not the reason the methods aren't static. This class depends on `MavenUtils` which is static and does actual Maven artifact resolution which I want to be able to mock in tests to avoid outgoing network connections.